### PR TITLE
Add `JobQueue.scheduler_configuration` and add corresponding warnings

### LINKED ISSUE #3837: [BUG] JobQueue.set_application drop jobstores field
### Steps to Reproduce

1. Configure JobQueue:
```python
job_queue = JobQueue()
job_queue.scheduler.configure(jobstores={'default': MongoDBJobStore()})
```
2. Build app:
```python
application = ApplicationBuilder().bot(tg_bot).job_queue(job_queue).build()
```

### Expected behaviour

`jobstores` should should not be dropped.

### Actual behaviour

After building JobQueue, jobstores are dropped.

### Operating System

linux

### Version of Python, python-telegram-bot & dependencies

```shell
python-telegram-bot 20.4 (2.0.3-1-g51cab98)
Bot API 6.7
Python 3.11.3 (main, Jun  5 2023, 09:32:32) [GCC 13.1.1 20230429]
```

### Relevant log output

_No response_

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by Bibo-Joshi**
Proposal for closing #3837

Makes the config values used by `JobQueue` publically accessible so that users can use them in addition to their custom values. Also adds warnigs that explain what this is about.

@rejedai, would you mind also having a look at this, given that you reported the issue?

- [x] Added ``.. versionadded:: NEXT.VERSION``, ``.. versionchanged:: NEXT.VERSION`` or ``.. deprecated:: NEXT.VERSION`` to the docstrings for user facing changes (for methods/class descriptions, arguments and attributes)
- [x] Created new or adapted existing unit tests
- [x] Documented code changes according to the `CSI standard <https://standards.mousepawmedia.com/en/stable/csi.html>`__
- ~Added myself alphabetically to ``AUTHORS.rst`` (optional)~
- ~Added new classes & modules to the docs and all suitable ``__all__`` s~
-  [x] Checked the `Stability Policy <https://docs.python-telegram-bot.org/stability_policy.html>`_ in case of deprecations or changes to documented behavior

closes #3837

### CODE REVIEW

File: telegram/ext/_jobqueue.py [hunk 1]
```diff
@@ -76,6 +76,17 @@ class JobQueue(Generic[CCT]):
...
         scheduler (:class:`apscheduler.schedulers.asyncio.AsyncIOScheduler`): The scheduler.
 
+            Warning:
+                This scheduler is configured by :meth:`set_application`. Additional configuration
+                settings can be made by users. However, calling
+                :meth:`~apscheduler.schedulers.base.BaseScheduler.configure` will delete any
+                previous configuration settings. Therefore, please make sure to pass the values
+                returned by :attr:`scheduler_configuration` to the method call in addition to your
+                custom values.
+                Alternatively, you cas also use methods like
```
Comment (lemontree210) [inline telegram/ext/_jobqueue.py:86]:
```suggestion
                Alternatively, you can also use methods like
```

File: telegram/ext/_jobqueue.py [hunk 2]
```diff
@@ -119,6 +128,43 @@ def application(self) -> "Application[Any, CCT, Any, Any, Any, JobQueue[CCT]]":
...
+            Since calling
+            :meth:`scheduler.configure() <apscheduler.schedulers.base.BaseScheduler.configure>`
+            deletes any previous setting, please make sure to pass these values to the method call
+            in addition to your custom values:
+
+            .. code-block:: python
+
+                scheduler.configure(..., **job_queue.scheduler_configuration)
+
+            Alternatively, you cas also use methods like
```
Comment (lemontree210) [inline telegram/ext/_jobqueue.py:145]:
```suggestion
            Alternatively, you can also use methods like
```

**Review by lemontree210 — CHANGES_REQUESTED** [review]
just a micro-typo

**Comment by lemontree210** [comment]
Please make sure you delete me from authors before merging, otherwise I'll be known for co-authoring a commit to master just because I changed one letter :)

**Review by harshil21 — COMMENTED** [review]
Changes look good to me. However codecov is complaining about untested lines. Is it really not possible to test them?

**Comment by harshil21** [comment]
macos py3.12 test consistently failing, but the errors aren't exactly related to this PR?

**Comment by Bibo-Joshi** [comment]
Mh, codecov doesn't complain [in their UI](https://app.codecov.io/gh/python-telegram-bot/python-telegram-bot/pull/3913/blob/telegram/ext/_jobqueue.py) and I'm also pretty certain that the tests cover everything. Maybe the annotations just stuck from the initial runs …

TBH I don't recall why the macos tests failed. Might have been due to codecov upload failure. I'm rerunning.

**Review by harshil21 — APPROVED** [review]
alright, test pass and code looks covered by the tests 👍🏽

**Pull request merged**